ecosystem/sep-0010.md

```diff
@@ -23,7 +23,9 @@ The authentication flow is as follows:
 1. The client obtains a unique [`challenge`](#challenge), which is represented as specially formed Stellar transaction
 1. The client verifies that the transaction has an invalid sequence number 0.  This is extremely important to ensure the transaction isn't malicious.
 1. The client verifies that the transaction is signed by the `SIGNING_KEY` specified by the requested service's [SEP-1 stellar.toml](sep-0001.md).
-1. The client verifies that the transaction includes a Manage Data operation, and that the value contains the home domain used to request the challenge.
+1. The client verifies that the transaction has a single Manage Data operation containing the user's account and a nonce.
+1. The client verifies that if the transaction has a single Manage Data operation with key `server_hostname` it contains the hostname of the SEP-10 server that the client requested the challenge.
+1. The client verifies that if the transaction has other Manage Data operations they all contain the server's account.
```

Comment:
I feel like the wording here might be a bit confusing – you could interpret as if there should be only one Manage Data operation, whereas we're actually expecting a few ones. Here's my suggestion:

```suggestion
1. The client verifies that the transaction has **exactly** one Manage Data operation containing the user's account and a nonce.
1. The client verifies the presence of **at most one** Manage Data operation with key `server_hostname` within the transaction. If present, the client verifies that this operation contains the hostname of the SEP-10 server from which the client requested the challenge.
1. The client verifies that if the transaction has other Manage Data operations they all contain the server's account.
```

Reply:
💯 I've modified it slightly here to remove the emphasis (bold) since there's lots of important text here and these aren't necessarily the more important than other word here. I also added a statement about the source account being set to the server account which is important.
```suggestion
1. The client verifies that the transaction has exactly one Manage Data operation containing the user's account and a nonce.
1. The client verifies the the transaction has at most one Manage Data operation with key `server_hostname` with the server's account as the source account. If present, the client verifies that this operation contains the hostname of the SEP-10 server from which the client requested the challenge.
1. The client verifies that if the transaction has other Manage Data operations they all contain the server's account.
```